### PR TITLE
Add multi github accounts to redhat-appstudio e2e tests

### PR DESCRIPTION
**Pull request by flacatus**
Jira link: https://issues.redhat.com/browse/STONE-565

### CODE REVIEW

File: ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-commands.sh [hunk 1]
```diff
@@ -7,20 +7,40 @@ set -o pipefail
...
+    -H "Authorization: Bearer ${GITHUB_USERNAME_ARRAY[1]}"\
+    https://api.github.com/rate_limit | jq ".rate.remaining")
+
+    if [[ "${GH_RATE_REMAINING}" -ge "${PREVIOUS_RATE_REMAINING}" ]];then
+        GITHUB_USER="${GITHUB_USERNAME_ARRAY[0]}"
+        GITHUB_TOKEN="${GITHUB_USERNAME_ARRAY[1]}"
+    fi
+    PREVIOUS_RATE_REMAINING="${GH_RATE_REMAINING}"
+    echo "[INFO ] user: $GITHUB_USER with rate limit remaining $GH_RATE_REMAINING"
+done
```
Comment (psturc) [inline ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-commands.sh:39]: shouldn't we have [separate tokens for HAS and e2e-tests](https://redhat-internal.slack.com/archives/C02FANRBZQD/p1676473620886019?thread_ts=1676472720.902939&cid=C02FANRBZQD)?
i.e.
`HAS_GITHUB_TOKEN` - would be used for bootstrapping the cluster (equivalent to `MY_GITHUB_TOKEN`)
`GITHUB_TOKEN` - used by e2e tests for sending requests to Github API

Comment (flacatus) [reply ci-operator/step-registry/redhat-appstudio/e2e/redhat-appstudio-e2e-commands.sh:39]: HAS do more requests to github than our tests... TBH i think will be better to have 3 users and balance the request between e2e and has

**Comment by flacatus** [comment]
cdq failed because i didnt add permissions to the new users to private repos 🎃

**Comment by flacatus** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests

**Comment by flacatus** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

**Comment by flacatus** [comment]
/pj-rehearse periodic-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests-periodic

**Comment by flacatus** [comment]
retriggered the 3 jobs after adding permissions

**Pull request merged**